HpToolsLauncher/Launcher.cs

```diff
@@ -121,19 +121,24 @@ public McConnectionInfo()
         public override string ToString()
         {
             string usernameOrClientId;
-            if (!string.IsNullOrEmpty(MobileClientId) && !string.IsNullOrEmpty(MobileSecretKey))
+            if (MobileAuthType == AuthType.AuthToken)
             {
                 usernameOrClientId = string.Format("ClientId: {0}", MobileClientId);
             }
             else
             {
-                usernameOrClientId = string.Format("Username: {0}", MobileSecretKey);
+                usernameOrClientId = string.Format("Username: {0}", MobileUserName);
             }
-            string McConnectionStr =
-                 string.Format("UFT Mobile HostAddress: {0}, Port: {1}, {2}, TenantId: {3}, UseSSL: {4}, UseProxy: {5}, ProxyType: {6}, ProxyAddress: {7}, ProxyPort: {8}, ProxyAuth: {9}, ProxyUser: {10}",
-                 MobileHostAddress, MobileHostPort, usernameOrClientId, MobileTenantId, GetMobileUseSslAsString(), GetMobileUseProxyAsString(), GetMobileProxyTypeAsString(), MobileProxySetting_Address, MobileProxySetting_Port, GetMobileProxyAuthenticationAsString(),
-                 MobileProxySetting_UserName);
-            return McConnectionStr;
+            string strProxy = string.Format("UseProxy: {0}", GetMobileUseProxyAsString());
+            if (MobileUseProxy == ONE)
+            {
+                strProxy += string.Format(", ProxyType: {0}, ProxyAddress: {1}, ProxyPort: {2}, ProxyAuth: {3}, ProxyUser: {4}", 
+                    GetMobileProxyTypeAsString(), MobileProxySetting_Address, MobileProxySetting_Port, GetMobileProxyAuthenticationAsString(), MobileProxySetting_UserName);
+            }
+            string mcConnStr =
+                 string.Format("UFT Mobile HostAddress: {0}, Port: {1}, AuthType: {2}, {3}, TenantId: {4}, UseSSL: {5}, UseProxy: {6}",

```

Comment:
`UseProxy: {6}`, the value of `{6}` is the variable `strProxy`, however, the `strProxy` value already begins with `UserProxy: ...`, we will get two `UseProxy` in the final string